libraries/stdlib/samples/test/samples/collections/collections.kt

```diff
@@ -773,6 +773,27 @@ class Collections {
             val emptyList = emptyList<Int>()
             assertPrints(emptyList.elementAtOrElse(0) { "no int" }, "no int")
         }
+
+        @Sample
+        fun find() {
+            val list = listOf(1, 2, 3, 1, 2, 3)
+            assertPrints(list.find { it == 1 }, "0")
```

Comment:
`find` return the found element and not its index in the list. Please, ensure that samples really pass by running `:kotlin-stdlib:samples:check` gradle task. 

Here, having the first and the last elements indistinguishable, it's hard to tell which one is actually returned.

So, I propose to use slightly less trivial predicate, that could return the same result for different elements, for example `it % 2 == 0` or `(it and 1) == 0` that checks parity.
```
val numbers = listOf(1, 2, 3, 4, 5, 6, 7)
val firstOdd = numbers.find { it % 2 != 0 }
val lastEven = numbers.findLast { it % 2 == 0 }
...
```